packages/preferences/src/components/preferences-modal/index.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+type ModalProps = Parameters< typeof Modal >[ 0 ];
+
+export type PreferencesModalProps = {
+	closeModal: ModalProps[ 'onRequestClose' ];
+	children: ReactNode;
```

Comment:
No react import needed

```suggestion

type ModalProps = Parameters< typeof Modal >[ 0 ];

export type PreferencesModalProps = {
	closeModal: ModalProps[ 'onRequestClose' ];
	children: React.ReactNode;
```